FirstClasses/fizzbuzz/src/fizzbuzz.erl

```diff
@@ -1,6 +1,35 @@
 -module(fizzbuzz).
 
--export([fizzbuzz/0]).
+-export([fizzbuzz/0, fizzbuzz2/0]).
 
+% I ran the tests but basically there is this assertion over small list of 5 elements.
+% The issue is that fizzbuzz is of arity 0 and the task asked to solve fizzbuzz for the 1..100 range.
+
+% Version 1
 fizzbuzz() ->
-    ok.
+    lists:map(
+        fun(Number) -> 
+           case Number of
+                _FizzBuzz when Number rem 3 == 0 andalso Number rem 5 == 0 ->
+                   'fizzbuzz';
+                _Fizz when Number rem 3 == 0 ->
+                   'fizz';
+                _Buzz when Number rem 5 == 0 ->
+                   'buzz';
+                _Else -> Number
+            end
+        end,
+        lists:seq(1,100)
+    ).
+
+% Version 2 ( after re-reading the documentation for Functions
```

Comment:
I must say version 2 is what I had in mind :)

Reply:
I know it is difficult to guess what I had in mind, I am just saying it is much much better :D

Reply:
Yeah , I needed some time to rap my head around this and figure out that this is just plain polymorphism :)